analysis.md

```diff
@@ -0,0 +1,77 @@
+
+## Layer 1
+The author of the script uses a lot of newline spacing to spread apart its contents. There is a function `ParsingA()` that appears to download content from the Internet, but it never seems to be used. You'll also notice that the author uses gratuitous string concatenation to confuse readers. For instance...
+
+```
+strComputer = "."
+Set objWMIService = GetObject("w" & "" & "in" & "" & "mgm" & "" & "ts" & "" & ":" & "" & "\\" & strComputer & "\r" & "" & "oot\c" & "" & "imv" & "" & "2")
+```
+...evaluates to the same thing as this: `Set objWMIService = GetObject("winmgmts:\\.\root\cimv2")`
+
+Also notice the large Base64 blob that `lStr` contains. The script decodes this blob and decompresses it for execution.
+
+### Unraveling the Base64
+Below is the Python3 script I use to decode the Base64 blob mentioned earlier.
+
+```
+import base64
+import sys
+
+ifile = open(sys.argv[1], "r")
+ofile = open(sys.argv[2], "w")
+
+encoded = ""
+
+for line in ifile:
+	encoded += line
+
+ofile.write(base64.b64decode(encoded))
+```
+After decoding, you'll notice that the resulting contents are still unreadable because they are gzipped. Decompress the contents using `gunzip` and you'll see an obfuscated powershell script with strange variable names meant to confuse readers. After cleaning up this mess through refactoring the code, there are many interesting things to note about malware persistence.
+
+The script uses various methods to keep the encoded third layer payload running on the victim machine regularly.
+
+## Layer 2
+
+### Storing the Payload
+If the victim has Powershell 3.0 or later, the script stores the encoded layer 3 payload inside `kernel32.dll`. Otherwise, the script adds a new property called `Path` to a key in the registry to store the payload. If the user has Administrator permissions, the payload will be stored in `HKLM:Software\Microsoft\Windows\CurrentVersion`. Otherwise, it will be stored in `HKCU:Software\Microsoft\Windows`.
+
+### Non-Admins: Registry Keys
+The script creates a file called `kernel32.vbs` and writes to it instructions to retrieve and execute the payload.
+
+The attacker modifies Run registry keys to cause `kernel.vbs` to run after waiting 30 minutes each time a user logs on. If the user has administrator permissions, the targeted key is `HKLM:Software\Microsoft\Windows\CurrentVersion\Run\`. Otherwise, the targeted key is `HKCU:Software\Microsoft\Windows\CurrentVersion\Run\`.
+
+### Admins: Creation of a New WMI Object
+If the user has administrator permissions, the script creates a permanent WMI event subscription that watches for a user logon, waits 30 minutes, then executes the layer 3 payload. At this point, the program has exhibited enough strange behavior for the SentinelAgent to identify it as malware.
```

Comment:
At this point, the layer 3 payload hasn't been introduced. What is the layer 3 payload? Is that the one which is downloaded by `kernel32.vbs` ?